sonar-web-plugin/src/main/java/org/sonar/plugins/web/lex/ElementTokenizer.java

```diff
@@ -154,8 +161,20 @@ private static void handleBeforeAttributeValue(CodeReader codeReader, TagNode el
         codeReader.pop();
         attribute.setQuoteChar((char) ch);
       } else {
-        codeReader.popTo(endTokenMatcher, sbValue);
-        attribute.setValue(sbValue.toString().trim());
+        char character;
+        while ((character = (char)codeReader.peek()) != -1) {
+          if (Character.isWhitespace(character)
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS134)